pkg/resource/deploy/step_generator.go

```diff
@@ -1890,6 +1914,18 @@ func (sg *stepGenerator) GenerateDeletes(targetsOpt UrnTargets, excludesOpt UrnT
 		steps = filtered
 	}
 
+	// TODO is this the right place to do this?
+	{
+		filtered := []Step{}
+		for _, step := range steps {
+			if step.Old() == nil || step.Old().ViewOf == "" {
```

Comment:
This is filtering _out_ view steps, right? I wonder if inverting the condition might be clearer, or maybe it's fine with a comment.